src/SocketWrapper.php

```diff
@@ -137,6 +137,7 @@ public function close()
         $this->loop->removeStream($this->fileDescriptor);
         $this->buffer->removeAllListeners();
         $this->removeAllListeners();
+        unset($this->buffer);
```

Comment:
Happy to hear this fixes the issues you were seeing! Unfortunately, this technically violates the above type hint (`Buffer|null`?) and also at least the `send()` method will try to access a non-existent property now. Can you look into this and possibly cover this with an additional test?

Reply:
Hi, I've put two checks to prevent this to happen in the SocketWrapper class.

Reply:
Nice, changes LGTM! :+1: Perhaps also update the type hint for `buffer` and `socket` to allow `null` values accordingly?